cng/cng.go

```diff
@@ -92,7 +92,7 @@ func utf16FromString(s string) []uint16 {
 func setString(h bcrypt.HANDLE, name, val string) error {
 	str := utf16FromString(val)
 	defer runtime.KeepAlive(str)
-	in := make([]byte, len(val)+1)
+	in := make([]byte, (len(val)+1)*2)
```

Comment:
This seems like a pretty significant fix--why now, and was aes usage of setString unaffected?

Reply:
Well, this is odd. Without this fix AES works as expected even though the chaining mode string buffer only contains half of the string. Even more, changing the part of the string that does not fit in the buffer make the test to fail, so the chaining mode is being correctly set. We don't have that luck with ECC curve names.

My guess is that CNG does not us the buffer length parameter when setting the chaining mode, it just look for the null terminated character. So, we haven't uncovered this latent bug until setting the ECC curve name, which probably checks the buffer length parameter.

Reply:
I've updated this function to make it simpler by using `unsafe.Slice`(https://pkg.go.dev/unsafe@master#Slice). In fact, `reflect.SliceHeader` will be deprecated since go1.19, so this is a move in the right direction.